view/app.js

```diff
@@ -1,14 +1,26 @@
 import { ethers } from 'ethers'
-const dpack = require('@etherpacks/dpack')
+const { getIpfs, providers } = require('ipfs-provider')
+const { httpClient, jsIpfs } = providers
 const dmap = require('../dmap.js')
 const dmapAddress = '0x7fA88e1014B0640833a03ACfEC71F242b5fBDC85'
 const dmapArtifact = require('../artifacts/core/dmap.sol/Dmap.json')
 
-window.onload =()=> {
+window.onload = async() => {
     const $ = document.querySelector.bind(document);
-    const result = $('#result')
-
     const line =s=> { $('#result').textContent += s + '\n' }
+    const node = await getIpfs({
+        providers: [
+            // attempt to use local node, if unsuccessful fallback to running embedded core js-ipfs in-page
+            httpClient({
+                loadHttpClientModule: () => require('ipfs-http-client'),
+                apiAddress: '/ip4/127.0.0.1/tcp/5001'
+            }),
+            jsIpfs({
+                loadJsIpfsModule: () => require('ipfs-core'),
```

Comment:
This is probably going to be overkill
Instead we probably want to just give the user a box to put their own rpc endpoint to use if they can't use one of a small set of default gateways
The important thing is to make sure the client actually verifies the CIDs when it grabs files from somewhere